apps/wallet/src/ui/app/hooks/useAccounts.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useMemo, useSyncExternalStore } from 'react';
+
+import { SyncedStore } from '../helpers/SyncedStore';
+
+import type { Account } from '_src/background/keyring/Account';
+
+type AccountType = ReturnType<Account['toJSON']>;
+
+export const accountsStore = new SyncedStore<AccountType[] | null>(null);
+
+export function useAccounts(addressFilter: string): AccountType | null;
+export function useAccounts(addressesFilter?: string[]): AccountType[];
```

Comment:
IMO function overloads are more confusing than they are useful, I'd just make this take an array and return an array always.